### PR TITLE
Do not crash when RawPTransform has null spec

### PR DESCRIPTION
**Pull request by kennknowles**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

R: @jkff or @tgroh 

Superficial issue: misuse of generated proto code. This is an obviously-correct fix that gets things green in the short term. I've confirmed this fix eliminates the NPE currently happening on `SplittableDoFnTest` in the Dataflow postcommit.

Larger issue: we stage a portable pipeline and the Dataflow job description has (or will have) pointers into that portable pipeline for the runner harness to extract portable payloads. So setting up `ProcessKeyedElements` to have a `null` payload or to be `NotSerializable` is incorrect. Instead, I think it has to be a standardized transform with a portable URN/payload specification. That will be a prerequisite to supporting SDF portably, unless we have a different implementation strategy.

### CODE REVIEW

**Comment by kennknowles** [comment]
run dataflow validatesrunner

**Comment by kennknowles** [comment]
I'll also add a unit test to catch it quickly

**Comment by kennknowles** [comment]
Added a test, rebased underneath and confirmed it crashes before and not after.

File: runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PTransformTranslation.java [hunk 1]
```diff
@@ -179,7 +179,10 @@ private PTransformTranslation() {}
     if (transform instanceof RawPTransform) {
       // The raw transform was parsed in the context of other components; this puts it in the
       // context of our current serialization
-      transformBuilder.setSpec(((RawPTransform<?, ?>) transform).migrate(components));
+      FunctionSpec spec = ((RawPTransform<?, ?>) transform).migrate(components);
+      if (spec != null) {
```
Comment (jkff) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PTransformTranslation.java]: Maybe add a brief comment why null specs are fine?

Comment (kennknowles) [reply runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PTransformTranslation.java]: TBH I'm on the fence about it. Just that GBK and Flatten could have dummy specs and we might as well save the typing and bandwidth...

Comment (kennknowles) [reply runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PTransformTranslation.java]: Sorry, I was conflating spec and payload. It is OK to have a null spec or a URN with a null payload.

**Pull request merged**